nautobot/extras/models/groups.py

```diff
@@ -431,3 +443,353 @@ def clean(self):
 
         # Validate `filter` dict
         self.clean_filter()
+
+    @classmethod
+    def generate_query_for_filter(cls, filter_field, value):
+        """
+        Return a `Q` object generated from the a `filter_field` and `value`.
+
+        :param filter_field:
+            Filter instance
+        :param value:
+            Value passed to the filter
+        """
+        q = models.Q()
+        if isinstance(filter_field, django_filters.MultipleChoiceFilter):
+            for v in value:
+                q |= models.Q(**filter_field.get_filter_predicate(v))
+        else:
+            q |= filter_field.get_filter_predicate(v)
+        return q
+
+    @classmethod
+    def generate_query_for_group(cls, group):
+        """
+        Return a `Q` object generated from all filters for a `group`.
+
+        :param group:
+            DynamicGroup instance
+        """
+        fs = group.filterset_class(group.filter, group.get_queryset())
+        q = models.Q()
+
+        for field_name, value in fs.data.items():
+            filter_field = fs.filters[field_name]
+            q &= cls.generate_query_for_filter(filter_field, value)
+
+        return q
+
+    def process_group_filters(self, group=None, qs=None):
+        """
+        Recursively process all filters for a dynamic group.
+
+        :param group:
+            DynamicGroup instance. If not provided, this group will be used.
+        :param qs:
+            Queryset to filter. If not provided, this group's queryset will be used.
+        """
+        if qs is None:
+            qs = self.get_queryset()
+
+        if group is None:
+            group = self
+            logger.debug("Processing group %s...", group)
```

Comment:
This is another one that has to be called recursively and since `get_queryset()` requires an instance (called on line 504), this must be an instancemethod. because the base queryset needs to be used to filter down. The base queryset is constructed from `content_type` which is only available on an instance.